operations/mimir/autoscaling.libsonnet

```diff
@@ -259,10 +311,20 @@
     if !$._config.autoscaling_ruler_querier_enabled then {} else removeReplicasFromSpec
   ),
 
+  ruler_query_frontend_scaled_object: if !$._config.autoscaling_ruler_query_frontend_enabled || !$._config.ruler_remote_evaluation_enabled then null else
+    newQueryFrontendScaledObject(
+      name='ruler-query-frontend',
+      cpu_requests=$.ruler_query_frontend_container.resources.requests.cpu,
+      memory_requests=$.ruler_query_frontend_container.resources.requests.memory,
+      min_replicas=$._config.autoscaling_ruler_query_frontend_min_replicas,
+      max_replicas=$._config.autoscaling_ruler_query_frontend_max_replicas,
+    ),
   ruler_query_frontend_deployment: overrideSuperIfExists(
     'ruler_query_frontend_deployment',
-    if !$._config.query_sharding_enabled || !$._config.autoscaling_ruler_querier_enabled then {} else
-      queryFrontendReplicas($._config.autoscaling_ruler_querier_max_replicas)
+    if $._config.autoscaling_ruler_query_frontend_enabled then removeReplicasFromSpec else
+      if $._config.query_sharding_enabled && $._config.autoscaling_ruler_querier_enabled then
```

Comment:
Use parenthesis here, otherwise it's hard to predict the precedence for the different `else` statements.

Reply:
Done, PTAL.